core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -26,24 +37,29 @@ use {
         clock::{Epoch, Slot, MAX_PROCESSING_AGE},
         fee::FeeBudgetLimits,
         saturating_add_assign,
-        transaction::SanitizedTransaction,
+        transaction::{MessageHash, SanitizedTransaction},
     },
     solana_svm::transaction_error_metrics::TransactionErrorMetrics,
-    std::sync::{Arc, RwLock},
+    solana_svm_transaction::svm_message::SVMMessage,
+    std::{
+        sync::{Arc, RwLock},
+        time::Instant,
+    },
 };
 
 pub(crate) trait ReceiveAndBuffer {
-    type Transaction: TransactionWithMeta;
-    type Container: StateContainer<Self::Transaction>;
+    type Transaction: TransactionWithMeta + Send + Sync;
+    type Container: StateContainer<Self::Transaction> + Send + Sync;
 
-    /// Returns whether the packet receiver is still connected.
+    /// Return Err if the receiver is disconnected AND no packets were
+    /// received. Otherwise return Ok(num_received).
     fn receive_and_buffer_packets(
         &mut self,
         container: &mut Self::Container,
         timing_metrics: &mut SchedulerTimingMetrics,
         count_metrics: &mut SchedulerCountMetrics,
         decision: &BufferedPacketsDecision,
-    ) -> bool;
+    ) -> Result<usize, ()>;
```

Comment:
This change was to improve the testing which was seeing some inconsistent failures, due to timing out in this call.

Those tests now loop until this call returns Ok(0) - or an error.

Natural question - Why did this only pop up for `TransactionView`?

The receive_and_buffer timeout for view is done differently than for sdk types.
The sdk types have a **receive** timeout, i.e. a time limit to receive packets, and then will deserialize all packets it received regardless of the time that takes.

The view type has an overall timeout so that it does not take longer than specified time for all operations - receive AND parsing.

Since receiving takes almost no time/work, this consistently would receive all the batches sent in the `sdk` variant. and then it would take however long to deserialize them - whether or not OS scheduler decides to rug the thread doesn't matter. If the view variant of the test gets rugged during parsing, then it can time out and not receive the next message.